.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,51 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - master
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  pull-requests: read
+  # Optional: Allow write access to checks to allow the action to annotate code in the PR.
+  checks: write
+
+jobs:
+  golangci:
+    strategy:
+      matrix:
+        go: ['1.21']
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
What's the benefit of running it on each OS type? Does it build it for that OS then lint?

Reply:
As I said [here](https://github.com/Boeing/config-file-validator/pull/138#issue-2236322411), I followed https://github.com/golangci/golangci-lint-action

As you are using gorelease to create binaries for multiple OS, testing them seemed to be a good thing.